src/public/js/util.js

```diff
@@ -286,7 +293,7 @@ var UTIL = (function() {
     for (var i = 0; i < polynomial.length; i++) {
       scaledPolynomial[i] = fieldMult(polynomial[i], scalar);
     }
-    return scaledPolynomial;
+    return trimLeadingZeros(scaledPolynomial);
```

Comment:
I think for this one you should trim on the parameter coming in, so var scaledPoly = trim(polynomial) then loop over scaledPoly multiplying each of its elements by scaler and replacing that element with the result.

If there are any leading 0s in result that weren't in the parameter other than if scaler is 0 something went horribly wrong.